pkg/logs/pipeline/pipeline.go

```diff
@@ -92,3 +82,45 @@ func (p *Pipeline) Flush(ctx context.Context) {
 	p.processor.Flush(ctx) // flush messages in the processor into the sender
 	p.sender.Flush(ctx)    // flush the sender
 }
+
+func getMainDestinations(endpoints *config.Endpoints, destinationsContext *client.DestinationsContext) *client.Destinations {
+	if endpoints.UseHTTP {
+		main := http.NewDestination(endpoints.Main, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend)
+		additionals := []client.Destination{}
+		for _, endpoint := range endpoints.GetUnReliableAdditionals() {
+			additionals = append(additionals, http.NewDestination(*endpoint, http.JSONContentType, destinationsContext, endpoints.BatchMaxConcurrentSend))
+		}
+		return client.NewDestinations(main, additionals)
+	}
+	main := tcp.NewDestination(endpoints.Main, endpoints.UseProto, destinationsContext)
+	additionals := []client.Destination{}
+	for _, endpoint := range endpoints.GetUnReliableAdditionals() {
+		additionals = append(additionals, tcp.NewDestination(*endpoint, endpoints.UseProto, destinationsContext))
+	}
+	return client.NewDestinations(main, additionals)
+}
+
+func getBackupDestinations(endpoints *config.Endpoints, destinationsContext *client.DestinationsContext) *client.Destinations {
```

Comment:
Would it be less confusing to tie things together from a readability perspective if these were `getReliableAdditionalDestinations()`? The less we have to guess and require diving deeper into what each of these functions _actually_ does the better.

Reply:
Good call - I renamed everything to be more consistent